src/c/tb_client.zig

```diff
@@ -63,7 +64,7 @@ else if (builtin.target.os.tag == .windows)
         var gpa = std.heap.HeapAllocator.init();
     }).gpa.allocator()
 else
-    @compileError("tb_client must be built with libc");
+    null;
```

Comment:
It's a bit hidden in the commit message now, but essentially `tb_client_header` would run (on linux, without libc) to generate the tb_client.h. And to do that, it would `@import("tb_client.zig")` for comptime inspection of its types. This may have eagerly evaluated `const global_allocator =` (as it needed to know the type of that decl) which caused the `@compileError` to be hit unexpectedly. 

This is my understanding of the problem at least.

Hiding the `@compileError` behind a decl to imply less strict evaluated didn't seem to help. In build.zig, I then tried adding `.linkLibC()` to the `tb_client_header` executable step, but this resulting in that glibc error I spoke of earlier today:
```bash
error(compilation): tigerbeetle/zig/lib/libc/glibc/sysdeps/x86_64/start-2.33.S:1:1: unable to build C object: FileNotFound
error(compilation): tigerbeetle/zig/lib/libc/glibc/csu/elf-init-2.33.c:1:1: unable to build C object: FileNotFound
error: unable to build glibc CRT file: BuildingLibCObjectFailed
```

Reply:
What if `global_allocator` was declared+defined in `init()`? (Or what is the reason for it to be "global"?)

Reply:
I thought that would work, butI finally realized that `tb_client_init_{echo}` being marked as `export fn` was the issue as it forced their functions (which used `global_allocator`) to be compiled/codegen'd into the `tb_client_header.zig` executable even if they weren't referenced. 

Removing the `export` basically fixes it, but im thinking of how to [conditionally export](https://ziglang.org/documentation/0.10.0/#export) it based on if it's apart of `tb_client_header` or `tb_client`.